services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/BandwidthSpec.groovy

```diff
@@ -68,6 +68,51 @@ class BandwidthSpec extends BaseSpecification {
         flow?.id && northboundService.deleteFlow(flow.id)
     }
 
+    def "Verify longer path is chosen in case of not enough available bandwidth on shorter path"() {
+        given: "Two active switches and two possible flow paths at least"
+        def switches = topology.getActiveSwitches()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .find { Switch src, Switch dst -> src != dst }
+        assert srcSwitch && dstSwitch
+
+        // Get possible flow paths and sort them by hops count.
+        def possibleFlowPaths = topologyEngineService.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path.sort { it.size() }
+        assert possibleFlowPaths.size() >= 2
+
+        // Make the first path more preferable than others.
+        possibleFlowPaths[1..-1].each { pathHelper.makePathMorePreferable(possibleFlowPaths.first(), it) }
+
+        // Get min available bandwidth on the preferable path.
+        def involvedBandwidths = []
+        def allLinks = northboundService.getAllLinks()
+        pathHelper.getInvolvedIsls(possibleFlowPaths.first()).each {
+            involvedBandwidths.add(islUtils.getIslInfo(allLinks, it).availableBandwidth)
+        }
+        def minAvailableBandwidth = involvedBandwidths.min()
+
+        // Create a flow to reduce available bandwidth on links of the preferable path.
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow1.maximumBandwidth = minAvailableBandwidth - 100
+        northboundService.addFlow(flow1)
+        def flow1Path = PathHelper.convert(northboundService.getFlowPath(flow1.id))
+        // Make sure that path of created flow is equal to the preferable path.
+        assert flow1Path == possibleFlowPaths.first()
+
+        when: "Create a flow. One path is shorter but available bandwidth is not enough, another path is longer"
+        def flow2 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow2.maximumBandwidth = 101
+        northboundService.addFlow(flow2)
+
+        then: "Flow is built through longer path"
+        def flow2Path = PathHelper.convert(northboundService.getFlowPath(flow2.id))
+        flow2Path != flow1Path
```

Comment:
I see that they are not equal. Why you say the the flow2Path is longer?

Reply:
Because we make all alternative paths not preferable (L83). All links of alternative paths have cost value set to 99999999.

Reply:
Ok, makes sense. Can we explicitly show this in the verification? Sth like
```
pathHelper.getCost(path2) > pathHelper.getCost(path1) 
```

Reply:
Getting a summary cost for a path/flow is really a tool we are missing in the framework atm